pkg/notaryv2/notaryv2.go

```diff
@@ -128,5 +136,118 @@ func (v *notaryV2Verifier) verifyOutcomes(outcomes []*notation.VerificationOutco
 }
 
 func (v *notaryV2Verifier) FetchAttestations(ctx context.Context, opts images.Options) (*images.Response, error) {
-	return nil, errors.Errorf("not implemented")
+	v.log.V(2).Info("fetching attestations", "reference", opts.ImageRef)
+
+	rcRepoReference, err := ref.New(opts.ImageRef)
+	if err != nil {
+		panic(err)
+	}
+
+	rcConfigHost := config.Host{
+		Name:     rcRepoReference.Registry,
+		Hostname: rcRepoReference.Registry,
+	}
+
+	rcClient := regclient.New(regclient.WithConfigHost(rcConfigHost))
+
+	rcRepoReferrers, err := rcClient.ReferrerList(ctx, rcRepoReference)
+	if err != nil {
+		msg := err.Error()
+		logger.Info("failed to fetch attestations", "error", msg)
+		if strings.Contains(msg, "MANIFEST_UNKNOWN: manifest unknown") {
+			return nil, fmt.Errorf("not found")
+		}
+
+		return nil, err
+	}
+	rcReferrersDescs := rcRepoReferrers.Descriptors
+	statements := make([]map[string]interface{}, 0)
+
+	certsPEM := combineCerts(opts)
+	certs, err := cryptoutils.LoadCertificatesFromPEM(bytes.NewReader([]byte(certsPEM)))
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse certificates")
+	}
+
+	trustStore := NewTrustStore("kyverno", certs)
+	policyDoc := v.buildPolicy()
+	notationVerifier, err := verifier.New(policyDoc, trustStore, nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to created verifier")
+	}
+
+	for _, referrer := range rcReferrersDescs {
```

Comment:
Do we need to verify the signatures or all matching predicates, or only the latest one in the list?

Reply:
the latest one is at the top so should I just break the loop when we have found the one?